src/deferred.js

```diff
@@ -304,6 +304,16 @@ jQuery.extend( {
 					// progress_callbacks.lock
 					tuples[ 0 ][ 2 ].lock
 				);
+
+				tuple[ 3 ].add(
```

Comment:
This doesn't look right to me; I think we should do everything in the existing handler (i.e., it should disable both `done` callbacks _and_ `then` callbacks, and lock both `done` _and_ then progress notifications). We could probably also stand to improve the comments, since "callbacks" is ambiguous in light of `done` vs. `then`.